src/cargo/core/compiler/build_context/target_info.rs

```diff
@@ -692,10 +691,36 @@ impl<'cfg> RustcTargetData<'cfg> {
         // `--target` flag is not specified. Since the unit_dependency code
         // needs access to the target config data, create a copy so that it
         // can be found. See `rebuild_unit_graph_shared` for why this is done.
-        if requested_kinds.iter().any(CompileKind::is_host) {
+        let target_applies_to_host = config.target_applies_to_host()?;
+        let host_config = if requested_kinds.iter().any(CompileKind::is_host) {
             let ct = CompileTarget::new(&rustc.host)?;
             target_info.insert(ct, host_info.clone());
-            target_config.insert(ct, host_config.clone());
+            let target_host_config = if target_applies_to_host {
+                let target_cfg_clone = config.target_cfg_triple(&rustc.host)?;
+                target_config.insert(ct, target_cfg_clone.clone());
+                target_cfg_clone
+            } else {
+                target_config.insert(ct, config.target_cfg_triple(&rustc.host)?);
+                config.host_cfg_triple(&rustc.host)?
+            };
+            target_host_config
+        } else {
+            if target_applies_to_host {
+                config.target_cfg_triple(&rustc.host)?
+            } else {
+                config.host_cfg_triple(&rustc.host)?
+            }
+        };
+
+        for kind in requested_kinds {
```

Comment:
This is new and not present before, how come this was necessary to add?

Reply:
Um, I think I botched the rebase on #9030, removed.